docs/website/01_framework/observer.md

```diff
@@ -26,38 +26,45 @@ enum class NonLinearSolverMessages {
 ## IObservable
 
 A class can be observable. The class then sends notifications when events are happening. To become observable, a class
-must inherit from `IObservable<MessageType>`, for example,
+must inherit from `IObservable<MessageType, InfoType>`, for example,
 
 ```cpp
-class NewtonRaphson : public IObservable<NonLinearSolverMessages> {...};
+class NewtonRaphson : public IObservable<NonLinearSolverMessages, NonLinearSolverLoggingInformation> {...};
 ```
 
-The function `this->notify(MessageType::Message)` is called at the appropriate position in the code to send a
-notification. This could be, for example,
+The function `this->notify(MessageType message, const InfoType& info)` is called at the appropriate position in the code
+to send a notification.
+This could be, for example,
 
 ```cpp
-this->notify(NonLinearSolverMessages::SOLUTION_CHANGED);
+this->notify(NonLinearSolverMessages::SOLUTION_CHANGED, logInfo);
 ```
 
 ## IObserver
 
-A class can be an observer. The class is then notified when events are happening and can perform actions. A very simple
-example is shown below. To become an observer, the class must inherit from ``IObserver<MessageType>``, where ``MessageType``
-is the `enum` of messages to use (see above).
+A class can be an observer.
+The class is then notified when events are happening and can perform actions.
+To become an observer, the class must inherit from ``IObserver<MessageType, InfoType>``.
+Here, ``MessageType``
+is the `enum` of messages to use (see above)
+and ``InfoType`` can either be ``NonLinearSolverLoggingInformation`` or ``ControlLoggerInformation``.
```

Comment:
Ifno type can be anything, since it is a template. It has to adhere to some interface, e.g. maybe a concept but now this is just some tempalte isn't it?